app/services/foreman_ansible/ansible_report_scanner.rb

```diff
@@ -5,19 +5,14 @@ module ForemanAnsible
   # sets the origin of the report to 'Ansible'
   class AnsibleReportScanner
     class << self
-      def scan(report, logs)
-        if (is_ansible = ansible_report?(logs))
-          report.origin = 'Ansible'
-        end
-        is_ansible
+      def add_reporter_data(report, raw); end
+
+      def identify_origin(raw)
+        raw['reporter']
```

Comment:
This could accidently return something else, in case another config mgmt tool would use this field, right?
Following seems bit safer:
```suggestion
        'Ansible' if ansible_report?(raw)
```

Reply:
Updated.